Kmeans_clustering_imgs.md

```diff
@@ -0,0 +1,204 @@
+
+Make a copy of the image we will cluster. This is in case we make modifications and want to preserve the original data.
+
+```
+img_init = img.copy() # save a copy of initial image, will modify this image for clustering
+```
+
+To view the image in its raw form:
+
+```
+plt.figure(figsize=(6, 6)) # plot initial image
+plt.imshow(img_init)
+```
+
+- We use *matplotlib*'s plotting function and *figsize=(6,6)* is just the size of our output viewing window in the colab notebook.
+
+
+It should appear exactly as expected.
+
+![](https://i.imgur.com/IltxwKj.png)
+
+
+
+Reshape the image above into a workable format, such as a 2-dimensional shape
+
+```
+img = img.reshape((img.shape[0] * img.shape[1],img.shape[2])) 
+```
+
+## Model Fitting
+Create an instance of the KMeans class. This basically means that grab the *KMeans* tool from our set of *sklearn* toolbox.
+
+Suppose that we'd like to extract 5 groups or colors from our dataset, pass in *n=5* as a parameter.
+```
+k = 5
+clt = KMeans(n_clusters = k) # "pick out" the K-means tool from our collection of algorithms
+clt.fit(img) # apply the model to our data, the image
+```
+
+Fitting the model means to apply our tool to our data, to familiarize our model with the contents of the data.
+
+
+## Handling the output
+The output of our model will need to be stored somehow. Create a numpy array that has length the number of clusters, starting at 0.
+
+
+Let's observe what the model outputs so far after fitting. Run:
+
+
+![](https://i.imgur.com/CZDhWav.png)
+
+- The first 3 datapoints belong to group *4* and the last 3 points belong to group *2*. Somewhere in between, there should be some assortment of 0s,1s, and 3s for the assigned group labels.
+
+
+
+To represent the number of colors we want to observe and in what quantity they appear in the image, a histogram will be used to portray the proportions.
+
+
+```
+label_indx = np.arange(0,len(np.unique(clt.labels_)) + 1) 
+```
+- initializes an array of length == # clusters, set the indices for the histogram
+
+Each "data" point of the image array will consist of its own color class label, so we plot the frequency of each color.The more a certain color appears in an image, the more data points it will have associated with it
+
+![](https://i.imgur.com/spfxuak.png)
+
+
+- since the output of this are two arrays and we only want to focus on the first one for the histogram, we use (_) to denote an empty storage, let the second array " go to nothing"
+
+Normalize the numbers within the array to get proportions that amount to 1. 
+
+![](https://i.imgur.com/f6vJfk4.png)
+
+- The proportions for each group (Group 0: 0.27, Group 1: 0.21...)
+
+Next, create a grid to hold our colors and their proportionate components. 
+
+```
+hist_bar = np.zeros((50, 300, 3), dtype = "uint8") 
+```
+- Initialize an array of some arbitary shape.
```

Comment:
why is some information in bullet points? for consistency, only use bullets when listing or something. It seems like most of the bullets could just be in sentences.